forloop_modules/utils/pickle_serializer.py

```diff
@@ -21,10 +24,17 @@ def save_data_dict_to_pickle_folder(data_dict, folder, clean_existing_folder=Tru
     os.makedirs(folder, exist_ok=True) #create folder if doesnt exist
 
     for k, v in data_dict.items():
+        # FIXME Ilya: serialization of BS objects hits recursion limit. Custom serializer is probably needed
+        if type(v) == bs4.element.Tag:
+            sys.setrecursionlimit(8000)
```

Comment:
please add description why this is necessary - why does it hit recursion limit? - nested tree?

Reply:
Yes, just due to the tree-like form of HTML